Add documentation for tsh proxy app (#10924)
gravitational/teleport: docs/pages/setup/reference/cli.mdx

$ tsh play --format=json ~/play/0c0b81ed-91a9-4a2a-8d7c-7495891a6ca0.tar | jq '.event
```

### tsh proxy app

> Start local TLS proxy for app connections when using Teleport in single-port mode

Reviewer: What happens if Teleport is not running in single-port mode when you run `tsh proxy app`?

> Start local TLS proxy for app connections when using Teleport in single-port mode

Reviewer: I also think there should be a sentence about why you would use this command vs. not using a local proxy for Application Access.

> Start local TLS proxy for app connections when using Teleport in single-port mode

Author: This command, as well as the other proxy subcommands seem to be a matter of convenience when working locally. Should I include a note on this?

> Start local TLS proxy for app connections when using Teleport in single-port mode

Author: The application works even not in single-port mode due to https being on 443. From my understanding reading the original PR for ALPN Proxy. The `proxy db` and `proxy ssh` will fall back on original connections when teleport is not running in single-port mode. I will attempt to confirm this. The original documentation for the existing commands before I added apps access was referencing single-port mode so I kept with that line.

> Start local TLS proxy for app connections when using Teleport in single-port mode

Reviewer: I think it would be great to add a sentence that this allows you to use CLIs for HTTP applications managed via Application Access without needing to log in separately for each command (if I'm understanding this feature correctly).

I also think we should confirm whether `tsh proxy app` requires TLS routing to work.

> Start local TLS proxy for app connections when using Teleport in single-port mode

Reviewer 2: > . While I'd like to document the addition I made, I think a potential followup issue might be best for clarifying the single-port mode comments that were originally made as well as documenting the other commands.

The other `tsh proxy` subcommands are already documented for example[ tsh proxy db](https://goteleport.com/docs/database-access/guides/gui-clients/#get-connection-information) and mared that this shoudl be used in `TLS routing` proxy mode.

> Start local TLS proxy for app connections when using Teleport in single-port mode

Author: > The other tsh proxy subcommands are already documented for example[ tsh proxy db](https://goteleport.com/docs/database-access/guides/gui-clients/#get-connection-information)

I was referring to the lack of reference within the [tsh cli reference](https://goteleport.com/docs/setup/reference/cli/#tsh)

```code
$ tsh proxy app [<flags>] <app>
```

#### Arguments
`<app>`

> - `app` The name of the application to start the local proxy for.

Reviewer: Should we specify that this needs to match the name of an application, e.g., as returned by `tsh apps ls`?

Author: Added a reference to this in https://github.com/gravitational/teleport/pull/10924/commits/2b9b9947be8a842d3f7deb415810eaebb1a14d75. Might need to be reworded but I think it makes it easier to understand.

Reviewer: Looks great, thanks

#### Flags

| Name | Default Value(s) | Allowed Value(s) | Description |
| - | - | - | - |
| `--port` | none | port | Specify the source port for the local proxy |

#### [Global Flags](#tsh-global-flags)

These flags are available for all commands `--login, --proxy, --user, --ttl, --identity, --cert-format, --insecure, --auth, --skip-version-check, --debug, --jumphost, --format`.
Run `tsh help <subcommand>` or see the [Global Flags Section](#tsh-global-flags)

#### Examples 
```code
$ tsh --proxy proxy.example.com:443 proxy app <app>

# Proxy a connection to drone app on local port 10700

> $ tsh proxy --port 10700 app drone

Reviewer 3: drone is probably a bit obscure.  I'd lean towards jenkins or grafana here.  Also, can we add a follup curl or api call to `127.0.0.1:10700` showing how to use this proxy?

Author: Do you mean the followup curl or API call as additional output from `tsh proxy --port 10700 app drone`. Or just documentation added here?

Author: Changed example from drone to grafana with https://github.com/gravitational/teleport/pull/10924/commits/271d714d4f51bbe6b7ff1988384fa90ce37c6955

Reviewer 3: > Do you mean the followup curl or API call as additional output from `tsh proxy --port 10700 app drone`. Or just documentation added here?

See this suggestion: https://github.com/gravitational/teleport/pull/10924#discussion_r822194739

> # Proxy a connection to grafana on local port 10700
> $ tsh proxy --port 10700 app grafana
> # Proxying connections to grafana on 127.0.0.1:10700

Reviewer 3: ```suggestion
# Proxy a connection to grafana on local port 10700
$ tsh proxy --port 10700 app grafana &
# Proxying connections to grafana on 127.0.0.1:10700
$ curl http://127.0.0.1:10700/api/users
```

$ tsh play --format=json ~/play/0c0b81ed-91a9-4a2a-8d7c-7495891a6ca0.tar | jq '.event
```

### tsh proxy app

> Start a local TLS proxy for Application Access connections when using Teleport in single-port mode.

Reviewer: ```suggestion
Starts a local TLS proxy for Application Access connections. 
You can use this proxy to connect to an application repeatedly after a single login to your Teleport cluster,
which is especially useful for interacting with an application via a CLI.
```

```code
$ tsh proxy app [<flags>] <app>
```

#### Arguments
`<app>`

- `app` The name of the application to start the local proxy for. To see a list of available applications, run `tsh apps ls`.

#### Flags

| Name | Default Value(s) | Allowed Value(s) | Description |
| - | - | - | - |

> | `--port` | none | port | Specify the source port for the local proxy |

Reviewer: ```suggestion
| `--port` | none | port number | Specify the source port for the local proxy |
```

#### [Global Flags](#tsh-global-flags)

These flags are available for all commands `--login, --proxy, --user, --ttl, --identity, --cert-format, --insecure, --auth, --skip-version-check, --debug, --jumphost, --format`.
Run `tsh help <subcommand>` or see the [Global Flags Section](#tsh-global-flags)

#### Examples 
```code

> $ tsh --proxy proxy.example.com:443 proxy app <app>

Reviewer 2: nit: the proxy address by default will be taken from take from `tsh` user profile. 
I don't see purpose of `--proxy proxy.example.com:443` flag in `tsh proxy app` and at first glance it make the command example harder to grasp.

Author: Agreed. Fixed in https://github.com/gravitational/teleport/pull/10924/commits/f3b68d2a1610cbfb32a9536471d1fcddccd8b899
